src/ai/backend/manager/container_registry/harbor.py

```diff
@@ -260,24 +260,26 @@ async def _scan_image(
                             if not image_info["tags"] or len(image_info["tags"]) == 0:
                                 skip_reason = "no tag"
                                 continue
-                            tag = image_info["tags"][0]["name"]
-                            match image_info["manifest_media_type"]:
-                                case self.MEDIA_TYPE_OCI_INDEX:
-                                    await self._process_oci_index(
-                                        tg, sess, rqst_args, image, tag, image_info
-                                    )
-                                case self.MEDIA_TYPE_DOCKER_MANIFEST_LIST:
-                                    await self._process_docker_v2_multiplatform_image(
-                                        tg, sess, rqst_args, image, tag, image_info
-                                    )
-                                case self.MEDIA_TYPE_DOCKER_MANIFEST:
-                                    await self._process_docker_v2_image(
-                                        tg, sess, rqst_args, image, tag, image_info
-                                    )
-                                case _ as media_type:
-                                    raise RuntimeError(
-                                        f"Unsupported artifact media-type: {media_type}"
-                                    )
+                            tags = [item["name"] for item in image_info["tags"]]
```

Comment:
The name field will always be present, right?

Reply:
I searched for relevant information in the Harbor API Swagger and API specification file but couldn't find any evidence that Tag.name is non-optional.
In the API specification below, we can find the definition of Tags, but the required array is not defined.
- https://github.com/goharbor/harbor/blob/main/api/v2.0/swagger.yaml

It’s just that, based on experience, there has never been a case where a Tag object didn’t have a name.